### PR TITLE
Pixel side scroller traps

### PR DESCRIPTION
**Pull request by l-seidl**
# Description
This PR adds spike trap objects and sounds to the pixel-side-scroller game.
- The sounds were made with jsfxr (so they are free to use)
- Also added a level to introduce the traps

spikes

## Type of change
- [ ] Bug fix (non breaking change that fixes an issue)
- [x] New feature (non breaking change that adds functionality)
- [ ] Tooling (non breaking change that adds functionality to the workflow)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)

# Checklist
- [x] My code follows the general style guidelines of the project
- [x] I have perfomed a self-review of my code
- [x] I have commented my code, particulary where it is unclear
- [x] My changes generate no new warnings or errors
- [x] The project compiles and runs correctly

### CODE REVIEW

File: game/games/pixel_side_scroller/objects/spikes.tscn [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+export(bool) var toggle_spikes = false
+export(bool) var initial_state = false
+export(float) var toggle_time_on = 1
+export(float) var toggle_time_off = 1
+export(float) var start_offset_time = 0
+
+
+# var _triggered: bool = false
+var _toggle_state = false
```
Comment (Joshix-1) [inline game/games/pixel_side_scroller/objects/spikes.tscn]: I don't like this name. If I look at this I don't know what true and what false means.
Maybe convert this to an enum that can be Spike or ground or just name it _is_currently_spike

File: game/games/pixel_side_scroller/pawns/character/character.gd [hunk 2]
```diff
@@ -9,7 +9,7 @@ func _process(_delta):
 
 
 func _animation() -> void:
-	if movement.direction.x == 0:
+	if movement.direction.x == 0 || is_on_floor() == false:
```
Comment (Joshix-1) [inline game/games/pixel_side_scroller/pawns/character/character.gd]: i think `if movement.direction.x == 0 or not is_on_floor():` would be better

**Review by Joshix-1 — COMMENTED** [review]
great work, but I still have a some suggestions

File: game/games/pixel_side_scroller/objects/spikes.tscn (hunk 1 above)
Comment (MarioLiebisch) [reply game/games/pixel_side_scroller/objects/spikes.tscn]: I kind of agree, "toggle state" is kind of like calling it "boolean". How about simply… `_active`?

Comment (RedstoneMedia) [reply game/games/pixel_side_scroller/objects/spikes.tscn]: `_active` is probably better than an enum

Comment (l-seidl) [reply game/games/pixel_side_scroller/objects/spikes.tscn]: Good Points :) will fix tomorrow when I have time

**Comment by MarioLiebisch** [comment]
Just as a minor note/suggestion: Don't describe your commit using `pr update`. Write what you changed, even if it's just 2-3 words, like `Updated misleading naming` or whatever, so people also instantly see what a commit tries to do. They can then later be squashed (so will disappear).

**Review by MarioLiebisch — APPROVED** [review]

File: game/games/pixel_side_scroller/objects/spikes.tscn [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+font_data = ExtResource( 1 )
+
+[sub_resource type="GDScript" id=5]
+script/source = "extends Label
+
+export(bool) var toggle_spikes = false
+export(bool) var initial_state = false
+export(float) var toggle_time_on = 1
+export(float) var toggle_time_off = 1
+export(float) var start_offset_time = 0
```
Comment (Numenter) [inline game/games/pixel_side_scroller/objects/spikes.tscn]: This removes the type mismatch warnings.
```suggestion
export(float) var toggle_time_on = 1.0
export(float) var toggle_time_off = 1.0
export(float) var start_offset_time = 0.0
```

**Pull request merged**